client/wfnews-war/src/main/angular/src/app/components/saved/saved.component.ts

```diff
@@ -86,9 +86,18 @@ export class SavedComponent implements OnInit {
         .subscribe((bans) => {
           this.savedLocations[outerIndex].bans = [];
           for (const innerIndex in bans?.features) {
-            const element = bans?.features[innerIndex];
-            this.savedLocations[outerIndex].bans.push(element);
-            this.cdr.markForCheck();
+            if (bans?.features.hasOwnProperty(innerIndex)) {
+              const element = bans?.features[innerIndex];
+          
+              // Check if the attribute already exists in the savedLocations array
+              const isAttributeAlreadyExists = this.savedLocations[outerIndex].bans.some(existingElement => 
```

Comment:
maybe `doesAttributeExist` is a better var name? Current one is bit of an weird mouthful. Could also "not" the attribute here so the `if` statement can be truthy